torch/csrc/jit/codegen/cuda/scheduler/normalization.cpp

```diff
@@ -268,6 +268,21 @@ std::shared_ptr<ReductionParams> innerPersistentHeuristic(
         (int64_t)vectorize_factor);
   }
 
+  // Set size of persistent per thread buffer on inner reduction buffer
+  int64_t batches_per_block_inner_reduction = 1;
+  int threads_after_vectorize =
+      inner_most_dimension_numel / inner_reduction_unroll_factor;
+  if (vectorize && threads_after_vectorize > max_threads_in_block) {
+    int warps_after_vectorize = ceilDiv(threads_after_vectorize, warp_size);
+    int warps_per_block = 32;
+    while (warps_per_block > 4 && warps_after_vectorize % warps_per_block) {
+      warps_per_block /= 2;
+    }
+    bdimx = warp_size * warps_per_block;
+  }
```

Comment:
I think I understand what and why you do this, but please leave comments and describe what the reasoning is.